mParticle-Apple-SDK/Next Gen/MPEventLogging.h

```diff
@@ -0,0 +1,50 @@
+//
+//  MPEventLogging.h
+//  mParticle-Apple-SDK
+//
+//  Created by Ben Baron on 6/29/23.
+//
+
+#import <Foundation/Foundation.h>
+#import "MPComponent.h"
+
+@class MPEvent;
+@class MPBaseEvent;
+@class MPCommerceEvent;
+@class MPNetworkPerformance;
+
+typedef NS_ENUM(NSUInteger, MPExecStatus) {
+    MPExecStatusSuccess = 0,
+    MPExecStatusFail,
+    MPExecStatusMissingParam,
+    MPExecStatusDisabledRemotely,
+    MPExecStatusEnabledRemotely,
+    MPExecStatusOptOut,
+    MPExecStatusDataBeingFetched,
+    MPExecStatusInvalidDataType,
+    MPExecStatusDataBeingUploaded,
+    MPExecStatusServerBusy,
+    MPExecStatusItemNotFound,
+    MPExecStatusDisabledInSettings,
+    MPExecStatusNoConnectivity
+};
+
+NS_ASSUME_NONNULL_BEGIN
+
+@interface MPEventLogging : NSObject <MPComponent>
+
+@property (nonatomic, strong, nullable) NSMutableSet<MPEvent *> *eventSet;
+
+- (void)leaveBreadcrumb:(nonnull MPEvent *)event completionHandler:(void (^ _Nonnull)(MPEvent * _Nonnull event, MPExecStatus execStatus))completionHandler;
+- (void)logError:(nullable NSString *)message exception:(nullable NSException *)exception topmostContext:(nullable id)topmostContext eventInfo:(nullable NSDictionary *)eventInfo completionHandler:(void (^ _Nonnull)(NSString * _Nullable message, MPExecStatus execStatus))completionHandler;
+- (void)logCrash:(nullable NSString *)message stackTrace:(nullable NSString *)stackTrace plCrashReport:(nonnull NSString *)plCrashReport completionHandler:(void (^ _Nonnull)(NSString * _Nullable message, MPExecStatus execStatus)) completionHandler;
+- (void)logBaseEvent:(nonnull MPBaseEvent *)event completionHandler:(void (^ _Nonnull)(MPBaseEvent * _Nonnull event, MPExecStatus execStatus))completionHandler;
```

Comment:
Your `logEvent` and `logBaseEvent` signatures look identical. Is there a major difference in the iOS SDK between `MPEvent` and `MPBaseEvent`?

Reply:
I think we should go for logEvent. If by any means let say we need to handle differently the event types (and eventually cast it or something), that shouldn't be exposed in implementation or naming, and should be transparent for the user. If you ask me, Im not a fan of MPBaseEvent either (which we also have on Android), our hierarchy and architecture shouldn't be exponsed.

Reply:
@alexs-mparticle MPEvent is a subclass of MPBaseEvent and internally MPEvents are logged via logBaseEvent. However see below regarding changes to these methods in this work:

@markvdouw For the scope of this work, we're not changing existing method signatures or business logic, only moving them. Later when we do more work toward the new API, we'll condense these into the simpler methods we discussed before (not having so many different methods to log events and adding a shim from the existing methods to the new, cleaner API).